### DISCUSSION TITLE
(M1 Mac) Cannot run game due to 2 exceptions (java.lang.UnsatisfiedLinkError) (java.lang.reflect.InvocationTargetException)

### DISCUSSION THREAD

**Discussion by drxjason** [wcko87/beatoraja-english-guide#41] [beatoraja questions / troubleshooting]
I can sucessfuly run the configuration menu but i cannot run the game itself, this comes up in my terminal when i click "Play"

[Pastebin containing exception since its too long](https://pastebin.com/nQNuNigr)

if your curious about `java -version`...
```
openjdk version "18.0.1.1" 2022-04-22
OpenJDK Runtime Environment (build 18.0.1.1+2)
OpenJDK 64-Bit Server VM (build 18.0.1.1+2, mixed mode, sharing)
```

checking `JAVA_HOME` using `/usr/libexec/java_home`...
```
/Library/Java/JavaVirtualMachines/liberica-jre-18-full.jre/Contents/Home
```
I do have several java installs but `JAVA_HOME` points to Liberica JRE 18
```
Matching Java Virtual Machines (5):
    18.0.1.1+2 (arm64) "BellSoft" - "BellSoft Liberica JRE 18.0.1.1+2" /Library/Java/JavaVirtualMachines/liberica-jre-18-full.jre/Contents/Home
    17.0.2 (arm64) "Oracle Corporation" - "Java SE 17.0.2" /Library/Java/JavaVirtualMachines/jdk-17.0.2.jdk/Contents/Home
    1.8.333.02 (x86_64) "Oracle Corporation" - "Java" /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Home
    1.8.0_333-b02 (arm64) "BellSoft" - "BellSoft Liberica JRE 1.8.0_333-b02" /Library/Java/JavaVirtualMachines/liberica-jre-8-full.jre/Contents/Home
    1.8.0_333-b02 (arm64) "BellSoft" - "BellSoft Liberica JDK 1.8.0_333-b02" /Library/Java/JavaVirtualMachines/liberica-jdk-8-full.jdk/Contents/Home
/Library/Java/JavaVirtualMachines/liberica-jre-18-full.jre/Contents/Home
```

I dont know if this is the right place to post this issue but any help would be appreciated

**Comment by wcko87** [comment]
This is the right place to ask.

I'm not exactly sure about the cause, but from the pastebin it looks like it might have to do with it failing to load the Discord RPC library on your OS.
There are few possible things you could try
- turning off Discord Rich Presence in the Other tab in the config
- setting up an older version of java (can be portable), try java 8 to be safest to see if that works (because I'm not sure how well beatoraja works with Java 18)
- try a slightly older version of beatoraja before they added the discord RPC (i.e. 0.8.3 or below) to see if that works

I wouldn't consider the third suggestion an actual solution, but it would be one way to isolate the cause of the issue.

**Reply by drxjason** [reply]
The first suggestion worked and also installed java 8 to be safe but... 
Now i get an error about `libgdx.dylib` failing to load
[Pastebin containing loading error](https://pastebin.com/3RSeQSAA)

**Reply by wcko87** [reply]
Maybe it has something to do with the program trying to load the 32-bit version of libgdx. I don't have a mac, so I can only guess from googling the error. Is there a reason why it's trying to load the 32-bit version? Are you running 32-bit java or have a machine that for some reason doesn't support 64-bit?

**Reply by drxjason** [reply]
AFAIK, 32-bit applications work somewhat out of the box, my architecture is 64-bit. My guess is that it's probably a compatibility issue with the library and or the game given the game was written to run on x86_64 systems or something. Regardless, Rosetta should be able to translate the application flawlessly. My machine runs on arm64/aarch64. The Java installs are ARM 64-bit installs. Either way I might look into the errors at another time, but thanks for your help anyways :)

**Reply by wcko87** [reply]
If you do find out anything and/or manage to resolve the issue, please do tell us about what you have found, since it would be great (at least for me) to have more information on any difficulties people may encounter when setting up beatoraja (especially on less frequently used OS's like OSX or Linux).

**Reply by wcko87** [reply]
> My machine runs on arm64/aarch64. The Java installs are ARM 64-bit installs. Either way I might look into the errors at another time, but thanks for your help anyways :)

By the way, try the x86 versions of Java, because there was someone else recently who had run into an error while running the ARM version of Java but things worked out fine after switching to x86 Java, despite having a machine that runs on arm64. They had a different error message though, which is why I didn't initially bring it up.

**Reply by drxjason** [reply]
> try the x86 versions of Java

So update!!! I installed the Intel 64-bit version of Liberica JRE and I can say it worked. Everything is running fine, ~~i havent tried any maps but it just runs.~~
